### PR TITLE
Packages step

### PR DESCRIPTION
**Pull request by lucasgarfield**
This is still very much a *draft* PR, it needs a lot of refactoring and there are probably still some bugs. However, the main functionality is all in place and I do not think that any of the logic will majorly change from this point. So, the only things left to do are to add unit tests and refactor.

### CODE REVIEW

**Comment by lucasgarfield** [comment]
Quick demo (also demos the really annoying bug we need to fix where the form resets to the first step):
cockpit packages draft pr

File: components/Wizard/CreateImageWizard.js [hunk 1]
```diff
@@ -23,8 +29,11 @@ const CreateImageWizard = (props) => {
     setIsWizardOpen(true);
   };
 
-  const handleSubmit = (formValues) => {
-    console.log(formValues);
+  const handleSubmit = async (formValues) => {
+    const result = await composer.diffBlueprintToWorkspace(props.blueprint.id);
```
Comment (jkozol) [inline components/Wizard/CreateImageWizard.js:33]: Let's just default commit. I don't want to compare against the workspace.

File: components/Wizard/CreateImageWizard.js [hunk 2]
```diff
@@ -5,9 +5,15 @@ import { Button } from "@patternfly/react-core";
...
 
 import ImageCreator from "./ImageCreator";
-import { imageOutput, details, review } from "./steps";
+import { imageOutput, details, packages, review } from "./steps";
 import "./CreateImageWizard.css";
 
+import BlueprintApi from "../../data/BlueprintApi";
+
+import * as composer from "../../core/composer";
+import { setBlueprint } from "../../core/actions/blueprints";
```
Comment (jkozol) [inline components/Wizard/CreateImageWizard.js:14]: This is unused. Did you intend on setting the blueprint before committing?

File: pages/blueprintEdit/index.js [hunk 3]
```diff
@@ -209,6 +209,8 @@ class EditBlueprintPage extends React.Component {
     // for now, just adding the component to the state
     // component info will load after committing the change to the workspace and reloading the blueprint
     const components = this.props.blueprint.components.concat(addedComponent);
+    console.log("blueprint: ", this.props.blueprint);
+    console.log("components, packages, modules, pendingChanges: ", components, packages, modules, pendingChanges);
```
Comment (jkozol) [inline pages/blueprintEdit/index.js:213]: Remove please.

**Review by jkozol — COMMENTED** [review]
The wizard refreshes when I add a package due to `Warning: Can't perform a React state update on an unmounted component`.

File: components/Wizard/CreateImageWizard.js (hunk 2 above)
Comment (lucasgarfield) [reply components/Wizard/CreateImageWizard.js:14]: Oof, good catch.

File: pages/blueprintEdit/index.js (hunk 3 above)
Comment (lucasgarfield) [reply pages/blueprintEdit/index.js:213]: Done.

**Comment by lucasgarfield** [comment]
Closing this PR as the commit diffs are not showing up correctly. The diff shows lines from a merge conflict that are not present in the feature branch on my private fork or on my local branch, so I think it must be a bug in Github.
image

**Pull request closed without merging**